Simplenote/src/main/java/com/automattic/simplenote/authentication/magiclink/MagicLinkConfirmationFragment.kt

```diff
@@ -0,0 +1,122 @@
+package com.automattic.simplenote.authentication.magiclink
+
+import android.content.Context
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.EditText
+import android.widget.TextView
+import android.widget.Toast
+import androidx.fragment.app.DialogFragment
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import com.automattic.simplenote.R
+import com.automattic.simplenote.authentication.SignInFragment
+import com.automattic.simplenote.authentication.SignUpCallback
+import com.automattic.simplenote.authentication.SimplenoteAuthenticationActivity
+import com.automattic.simplenote.viewmodels.CompleteMagicLinkViewModel
+import com.automattic.simplenote.viewmodels.MagicLinkUiState
+import com.simperium.android.ProgressDialogFragment
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class MagicLinkConfirmationFragment : Fragment() {
+
+    private var progressDialogFragment: ProgressDialogFragment? = null
+
+    private var codeEditText: EditText? = null
```

Comment:
Wonder if those View variables could be local?

Reply:
Yes to `codeEditText`. But no to `progressDialogFragment`. The reason being is we show and hide it from multiple places.

Reply:
Yes, I only meant the views - codeEditText, actionCodeButton and loginWithPassword in this case:)